src/main/java/com/google/sps/data/DatastoreClient.java

```diff
@@ -0,0 +1,136 @@
+    * return Optional is empty, then no object exists with given parameter. 
+    */  
+  public Optional<Instance> getInstance(String instanceName) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query getInstance = new Query(EntityConstants.InstanceEntity.TABLE_NAME)
+      .setFilter(new FilterPredicate
+      (EntityConstants.InstanceEntity.INSTANCE_NAME, 
+      FilterOperator.EQUAL, instanceName));
+    PreparedQuery pq = datastore.prepare(getInstance);
+    Entity instanceEntity = pq.asSingleEntity();
+    return Optional.of(Instance.fromEntity(instanceEntity));
+   }
+
+  /** 
+   * Returns the Attendee object associated with given screenName. If 
+   * return Optional is empty, then no object exists with given parameter.
+   */  
+  public Optional<Attendee> getAttendee(String screenName) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query getAttendee = new Query(EntityConstants.AttendeeEntity.TABLE_NAME)
+      .setFilter(new FilterPredicate
+      (EntityConstants.AttendeeEntity.SCREEN_NAME,
+      FilterOperator.EQUAL, screenName));
+    PreparedQuery pq = datastore.prepare(getAttendee);
+    Entity attendeeEntity = pq.asSingleEntity();
+    return Optional.of(Attendee.fromEntity(attendeeEntity));
+   }
+
+  /** Deletes an attendee from the datastore. */
+  public void deleteAttendee(String screenName) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query getAttendee = new Query(EntityConstants.AttendeeEntity.TABLE_NAME)
+      .setFilter(new FilterPredicate
+      (EntityConstants.AttendeeEntity.SCREEN_NAME,
+      FilterOperator.EQUAL, screenName));
+    PreparedQuery pq = datastore.prepare(getAttendee);
+    Entity attendeeEntity = pq.asSingleEntity();
+    datastore.delete(attendeeEntity.getKey());
+    }
+
+  /** Returns a list of attendees in a session. */
+  public List<Attendee> getAttendeesInSession
+    (String sessionId) {
+      DatastoreService datastore = 
+        DatastoreServiceFactory.getDatastoreService();
+      Query query = new Query(EntityConstants.AttendeeEntity.TABLE_NAME)
+        .setFilter(new FilterPredicate
+        (EntityConstants.AttendeeEntity.SESSION_ID, FilterOperator.EQUAL, 
+        sessionId));
+      PreparedQuery results = datastore.prepare(query);
+      List<Attendee> attendeeList = new ArrayList<>();
+      for (Entity attendeeEntity : results.asIterable()) {
+        attendeeList.add(Attendee.fromEntity(attendeeEntity));
+      }
+      return attendeeList;
+  }
+
+  /** Returns a list of available instance. */
+  public List<Instance> getAvailableInstances() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query(EntityConstants.InstanceEntity.TABLE_NAME)
+      .setFilter(new FilterPredicate(EntityConstants.InstanceEntity.STATE, 
+      FilterOperator.EQUAL, "TERMINATED"));
```

Comment:
Create a private final for TERMINATED.

private final String INSTANCE_STATE_TERMINATED = "TERMINATED";

That should be good for now. I think later we'll actually want a full enum but we'll cross that bridge when we get there.